_internal/types.ts

```diff
@@ -14,15 +22,24 @@ export type FetcherResponse<Data = unknown> = Data | Promise<Data>
 export type BareFetcher<Data = unknown> = (
   ...args: any[]
 ) => FetcherResponse<Data>
+/**
+ * Second parameter provided to fetcher functions. At present this only allows
+ * for 'signal' for cancelling a request in progress, but allows for future
+ * enhancements.
+ */
+export interface FetcherOptions {
+  signal?: AbortSignal
+  [key: string]: unknown
```

Comment:
I'd omit an index signature because it may hide typos for the valid options (only `signal` in this case). It also prevents us from adding properties without introducing a potential breaking change.